rest/views.py

```diff
@@ -411,9 +417,20 @@ class LearningResourceList(ListAPIView):
 
     def get_queryset(self):
         """Get queryset for a learning resource"""
-        return LearningResource.objects.filter(
+        queryset = LearningResource.objects.filter(
             course__repository__slug=self.kwargs['repo_slug']
         )
+        id_value = self.request.query_params.get('id', None)
+        if id_value is not None:
+            if len(id_value) == 0:
```

Comment:
This check for zero is unnecessary. The list comprehension in the `else` will return an empty list if `if len(x) > 0` is added and will handle both cases.